src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -269,6 +269,9 @@ void QgsRelationReferenceWidget::setForeignKey( const QVariant &value )
 
       for ( int i = 0; i < mFilterFields.size(); i++ )
       {
+        if ( i >= mFilterComboBoxes.size() )
```

Comment:
Would it make sense to just replace `mFilterFields.size()` one line further up?

Reply:
We could but we should check that `i` doesn't exceed the size of `mFilterFields` before doing `QVariant v = mFeature.attribute( mFilterFields[i] );` then.

But maybe I'm missing your point?

Reply:
I wasn't sure if it is guaranteed that `mFilterComboBoxes.size() >= mFilterFields.size()`, but I didn't check.

If not, I'd probably do a `int field_count = min( mFilterComboBoxes.size(), mFilterFields.size() )` which doesn't require any jumps like `break;`.

Reply:
Good point, I'm going to do that. Thanks for the advice!

Reply:
Done